helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -225,10 +227,52 @@ protected StatefulRebalancer createRebalancer(HelixManager helixManager) {
    *  1) ZK callback should go to ClusterDataCache and trigger data cache refresh only
    *  2) then ClusterDataCache.refresh triggers rebalance pipeline.
    */
-  /* Map of cluster->GenrichelixController */
-  private static Map<String, GenericHelixController> HelixControllerFactory = new ConcurrentHashMap<>();
-  public static GenericHelixController getController(String clusterName) {
-    return HelixControllerFactory.get(clusterName);
+  /* Map of cluster->Set of GenericHelixController*/
+  private static Map<String, ImmutableSet<GenericHelixController>> _helixControllerFactory =
```

Comment:
minor: for static variables, what is our name convention?  I think _wordAnotherWord format is for instance variable?

Reply:
The constants (static final) should be all upper case + underscore. But this var is not constants. Then, all the other related naming conventions include,
1. Class and instance variables: _fooBar (this is the one that makes sense)
2. Local variables: fooBar

On the other hand, HelixControllerFactory is not a good option. Since it conflicts with the Classes/Interfaces naming convention.